File: src/main/java/fyordo/lifeagragator/med/tag/request/TagUpdateRequest.java
package fyordo.lifeagragator.med.tag.request;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.NonNull;

@EqualsAndHashCode(callSuper = true)
@Data
@NoArgsConstructor
public class TagUpdateRequest extends TagCreateRequest {
    @NonNull
    private Long id;

    public TagUpdateRequest(@NonNull Long id, @NonNull String title, @NonNull String color, @NonNull String textColor) {
        super(title, color, textColor);
        this.id = id;
    }
}

File: src/main/kotlin/fyordo/lifeagragator/med/tag/dto/TagUpdateRequest.kt
package fyordo.lifeagragator.med.tag.dto

import fyordo.lifeagragator.med.base.utils.getUserId

data class TagUpdateRequest(
    var id: Long,

    val title: String,

    val color: String,

    val textColor: String,

    val createdUserId: Long? = getUserId()
)
